### PR TITLE
Extract fragments into package

### PR DESCRIPTION
**Pull request by BlakeWilliams**
The primary driver behind this PR is around clarifying naming. Previously we had `multiplexer.fragment` and `viewproxy.Fragment` and it wasn't always clear when you're using one or the other. This attempts to clarify the naming by extracting `multiplexer.fragment` into a `Requestable` interface and `viewproxy.Fragment` into a new package/struct, `package.Fragment`.

* Extract fragment logic into a small `fragment` package
* Rename fragment type to `fragment.Definition` to be more clear about its
  intended use
* Replace `multiplexer` package's `fragment` type with a `Requestable`
  interface
* Implement `IntoRequestable` on `fragment.Definition` so it can be used with
  the `multiplexer` package.
* Remove the need to explicitly sort in the multiplexer

### CODE REVIEW

File: pkg/multiplexer/multiplexer.go [hunk 1]
```diff
@@ -25,7 +24,7 @@ type Request struct {
 	ctx          context.Context
 	Header       http.Header
 	layoutURL    string
-	fragments    []Fragment
+	fragments    []Requestable
```
Comment (iancanderson) [inline pkg/multiplexer/multiplexer.go:27]:
```suggestion
	requestables    []Requestable
```

maybe multiplexer doesn't need to have a concept of "fragments"?

File: pkg/multiplexer/multiplexer.go [hunk 2]
```diff
@@ -25,7 +24,7 @@ type Request struct {
 	ctx          context.Context
 	Header       http.Header
 	layoutURL    string
```
Comment (iancanderson) [inline pkg/multiplexer/multiplexer.go:26]:
```suggestion
```

is this unused?

File: pkg/fragment/fragment.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"net/url"
+	"strings"
+
+	"github.com/blakewilliams/viewproxy/pkg/multiplexer"
+)
+
+type Collection = []*Definition
+type DefinitionOption = func(*Definition)
+
+type Definition struct {
```
Comment (iancanderson) [inline pkg/fragment/fragment.go:14]: maybe rename this file to `definition.go`? 🤷

File: pkg/middleware/logging/logging.go [hunk 4]
```diff
@@ -74,11 +74,12 @@ func (t *logTripper) Request(r *http.Request) (*http.Response, error) {
 	start := time.Now()
 	res, err := t.tripper.Request(r)
 	duration := time.Since(start)
-	fragment := viewproxy.FragmentFromContext(r.Context())
+	fragment := multiplexer.RequestableFromContext(r.Context())
```
Comment (iancanderson) [inline pkg/middleware/logging/logging.go:77]:
```suggestion
	requestable := multiplexer.RequestableFromContext(r.Context())
```

File: pkg/middleware/logging/logging.go [hunk 5]
```diff
@@ -74,11 +74,12 @@ func (t *logTripper) Request(r *http.Request) (*http.Response, error) {
 	start := time.Now()
 	res, err := t.tripper.Request(r)
 	duration := time.Since(start)
-	fragment := viewproxy.FragmentFromContext(r.Context())
+	requestable := multiplexer.RequestableFromContext(r.Context())
```
Comment (BlakeWilliams) [inline pkg/middleware/logging/logging.go:77]: This was previously returning a `multiplexer.fragment`, not a `viewproxy.Fragment` so this should be equivalent

**Review by iancanderson — COMMENTED** [review]
This is looking good! Is the CI failure flakey or related?

**Comment by BlakeWilliams** [comment]
Unfortunately it looks like it's a flake. I'll re-run for now but investigate the issue later.

**Pull request merged**